internal/server/evaluation/evaluation.go

```diff
@@ -0,0 +1,84 @@
+package evaluation
+
+import (
+	"context"
+	"errors"
+
+	errs "go.flipt.io/flipt/errors"
+	fliptotel "go.flipt.io/flipt/internal/server/otel"
+	"go.flipt.io/flipt/internal/storage"
+	"go.flipt.io/flipt/rpc/flipt"
+	rpcEvaluation "go.flipt.io/flipt/rpc/flipt/evaluation"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"go.uber.org/zap"
+)
+
+// Variant evaluates a request for a multi-variate flag and entity.
+func (s *Server) Variant(ctx context.Context, v *rpcEvaluation.EvaluationRequest) (*rpcEvaluation.VariantEvaluationResponse, error) {
+	var ver = &rpcEvaluation.VariantEvaluationResponse{}
+
+	flag, err := s.store.GetFlag(ctx, v.NamespaceKey, v.FlagKey)
+	if err != nil {
+		var errnf errs.ErrNotFound
+
+		if errors.As(err, &errnf) {
+			ver.Reason = rpcEvaluation.EvaluationReason_FLAG_NOT_FOUND_EVALUATION_REASON
+			return ver, err
+		}
+
+		ver.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+		return ver, err
+	}
+
+	if flag.Type != flipt.FlagType_VARIANT_FLAG_TYPE {
+		ver.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
```

Comment:
We're going to stop adding reasons when we return non-nil errors.
I validated and gRPC replaces the body of the response with the error on the wire anyways.
So the whole response is nil from the client perspective, making reason moot.

For the same reason we're going to drop FLAG_NOT_FOUND_REASON altogether.

Reply:
@GeorgeMac Definitely. That makes sense.